packages/data-service/src/provider.ts

```diff
@@ -5,6 +5,11 @@ const DataServiceContext = createContext<DataService | null>(null);
 
 export const DataServiceProvider = DataServiceContext.Provider;
 
+export type DataServiceLocator<
+  K extends keyof DataService = keyof DataService,
+  L extends keyof DataService = K
+> = () => Pick<DataService, K> & Partial<Pick<DataService, L>>;
```

Comment:
@gribnoysup heads up, I added this because it seemed that `depcheck` cannot actually process the `typeof dataServiceLocator<...` syntax so it gave invalid results

A bit unfortunate that we’re still in a place where depcheck and eslint are holding back our ability to adopt TS features but I also don’t want to open that can of worms right now :)

Reply:
Yeah, definitely a bit annoying that this is what's causing it, but generally not bad at all to have this as a separate type that we can import 👍